js/future.md

```diff
@@ -0,0 +1,67 @@
+
+The only focus so far in development has been on correctness, and no effort has yet gone into performance optimization.
+
+Due to the design of the MLT specification, with its flexibility and features, major opportunities exist to improve performance, as listed below:
+
+## Advanced Encodings
+
+|Performance Estimate| Effort Estimate|
+|--------------------|----------------|
+| 20-30 ops/s        | Moderate (days)|
+
+One of the most novel aspects of the MLT specification is the efficiency gained in combining column-oriented data with lightweight encodings tailored to the data type. To take advantage of this combination, the MLT spec supports what we refer to as "advanced encodings," which are mainly FastPFor and FSST.
+
+However, the JS decoder has not yet attempted to support these, such that the decoding performance benefits have not been realized, and the input tiles are not yet as small as they could be.
+
+Work on this is not yet planned, but it is tracked at https://github.com/maplibre/maplibre-tile-spec/issues/222.
+
+## Lazy Geometry Decoding
+
+|Performance Estimate| Effort Estimate|
+|--------------------|----------------|
+| 30-70 ops/s        | Moderate (days)|
+
+Another novel aspect of the MLT specification is how feature-rich its geometry storage structure is. One of the features of its storage structure allows for on-demand (aka lazy) decoding of geometries. This allows a parser to iterate all features in a layer and only pay the cost of materializing geometries if needed and right when needed.
+
+However, the JS decoder does not yet support this feature. Work on this is not yet planned, but it is tracked at https://github.com/maplibre/maplibre-tile-spec/issues/224.
+
+## Reducing Memory Allocations
+
+|Performance Estimate| Effort Estimate|
+|--------------------|----------------|
+| 10-25 ops/s        | Moderate (days)|
+
+A large amount of array allocation is needed by the current JS decoder to translate the current column-oriented design of an MLT into the row-oriented format expected by MapLibre. So, we can achieve very easy performance improvements by optimizating this array allocation in obvious ways like pre-allocating when the final size is know and using TypedArrays over Array when allocation speed of the former is more effecient the the often slower access is not critical.
+
+But the bigger picture issue here is that too much time spent optimizing this may not be required because a bigger performance win awaits the developer who can effecitvely refactor MapLibre to work directly off of the column-oriented data. This would allow MapLibre to efficiently access data within a column without allocating the entire column of data in order to assemble a fully materialized feature. For example, the current MapLibre code expects fully materialize features with all their properties available in the expression evaluation/filtering path. It could be a big but 1. Optimizing text decoding
```

Comment:
Is this a typo?

```suggestion
But the bigger picture issue here is that too much time spent optimizing this may not be required because a bigger performance win awaits the developer who can effecitvely refactor MapLibre to work directly off of the column-oriented data. This would allow MapLibre to efficiently access data within a column without allocating the entire column of data in order to assemble a fully materialized feature. For example, the current MapLibre code expects fully materialize features with all their properties available in the expression evaluation/filtering path.
```